src/components/zenlytic-legend/ZenlyticLegend.js

```diff
@@ -16,39 +18,25 @@ const ZenlyticLegend = ({
   useStrokeColorShape,
   ...restProps
 }) => {
-  const legendItem = (value, entry) => {
-    const strokeColor = entry?.payload?.stroke;
+  const renderLegend = (props) => {
```

Comment:
This looked to be the best way to achieve the "if more than N legend items, only show N" logic from the API for their legend (link: https://recharts.org/en-US/api/Legend).